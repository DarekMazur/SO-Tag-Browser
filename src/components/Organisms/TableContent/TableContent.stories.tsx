import type { Meta, StoryObj } from '@storybook/react';
import TableContent from './TableContent.tsx';

const meta: Meta<typeof TableContent> = {
  tags: ['autodocs'],
  component: TableContent,
  title: 'Organisms/TableContent',
};

export default meta;

type Story = StoryObj<typeof TableContent>;

export const Default: Story = {};
