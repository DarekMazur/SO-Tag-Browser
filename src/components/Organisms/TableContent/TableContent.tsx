import { TableBody, TableRow } from '@mui/material';
import { useSelector } from 'react-redux';
import { mockTags } from '../../../mocks/mocks.ts';
import TagLink from '../../Atoms/TagLink/TagLink.tsx';
import TableItem from '../../Atoms/TableItem/TableItem.tsx';

const TableContent = () => {
  const pagination = useSelector((state) => state.pagination);
  const tagsPerPage = useSelector((state) => state.tagsPerPage);

  return (
    <TableBody>
      {(tagsPerPage > 0
        ? mockTags.items.slice(
            pagination * tagsPerPage,
            pagination * tagsPerPage + tagsPerPage,
          )
        : mockTags.items
      ).map((tag, index) => (
        <TableRow key={tag.name}>
          <TableItem label={index + 1 + pagination * tagsPerPage} />
          <TableItem label={tag.name} />
          <TableItem label={new Intl.NumberFormat().format(tag.count)} />
          <TableItem label={<TagLink to={tag.name} />} />
        </TableRow>
      ))}
    </TableBody>
  );
};

export default TableContent;
