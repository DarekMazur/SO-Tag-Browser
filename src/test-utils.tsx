// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import AppProviders from './providers/AppProviders.tsx';

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <AppProviders>{children}</AppProviders>;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customRender = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

export { customRender as render };
