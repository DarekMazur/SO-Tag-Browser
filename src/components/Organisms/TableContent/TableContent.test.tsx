import { render, screen } from '../../../test-utils';
import TableContent from './TableContent.tsx';

describe('Search Bar', () => {
  it('Renders the component', () => {
    const { container } = render(<TableContent />);
    expect(container).toMatchSnapshot();
  });
  it('Displays table items from API', async () => {
    render(<TableContent />);
    screen.findAllByText(/Loading/);
    await screen.findAllByText(/javascript/);
  });
});
