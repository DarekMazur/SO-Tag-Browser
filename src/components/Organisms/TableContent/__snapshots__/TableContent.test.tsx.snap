// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Bar > Renders the component 1`] = `
<div>
  <tbody
    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  />
</div>
`;
